modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js

```diff
@@ -14,65 +14,141 @@
 
 import ClayLayout from '@clayui/layout';
 import classNames from 'classnames';
-import React, {useContext, useRef} from 'react';
+import {DragTypes} from 'data-engine-taglib';
+import {useEventListener} from 'frontend-js-react-web';
+import React, {
+	forwardRef,
+	useContext,
+	useEffect,
+	useRef,
+	useState,
+} from 'react';
+import {useDrag} from 'react-dnd';
+import {getEmptyImage} from 'react-dnd-html5-backend';
 
+import {EVENT_TYPES} from '../../actions/eventTypes.es';
 import {DND_ORIGIN_TYPE, useDrop} from '../../hooks/useDrop.es';
+import {useForm} from '../../hooks/useForm.es';
 import {hasFieldSet} from '../../util/fields.es';
 import {Actions, ActionsControls, useActions} from '../Actions.es';
 import {ParentFieldContext} from '../Field/ParentFieldContext.es';
 import {Placeholder} from '../Placeholder.es';
 import * as DefaultVariant from './DefaultVariant.es';
 
+const DIRECTIONS = {
+	LEFT: 'left',
+	RIGHT: 'right',
+};
+
+const MAX_COLUMNS = 12;
+
+const FieldDragPreview = ({containerRef}) => {
+	const ref = useRef(null);
+
+	/**
+	 * This hack was needed to capture the field snapshot.
+	 * Currently the Field is loaded lazily and the preview
+	 * will look like a loading state field.
+	 */
+	useEffect(() => {
+
+		/**
+		 * It copies the width of the field and clone the DOM element
+		 * to replace the ref inner FieldDragPreview
+		 */
+		const {width} = getComputedStyle(containerRef.current);
+		const element = containerRef.current.cloneNode(true);
+
+		ref.current.parentElement.style.width = width;
+
+		ref.current.appendChild(element);
+
+		return () => {
+			ref.current.remove();
+		};
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
SF still failing here but I used correctly `eslint-disable-next-line`. I expect when rebuilding entire DXP will solve this on my machine.

Reply:
Solved on e7db112 Thx Matu <3